### PR TITLE
tests/operator, test_id:3145: Replace Update with Patch

### PR DESCRIPTION
**Pull request by RamLavi**
### What this PR does
It is common practice on our e2e tests to use Update k8s API in order to run the test. In order to ensure the change was executed, these also are wrapped with Eventually/RetryOnConflict.
The issue is that most often tests do not really need to use Update (for an atomic change) and Patch would be quicker and more precises.
This PR is replacing an unneeded use of update in favor of patch, in test_id:3145.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/operator/operator.go [hunk 1]
```diff
@@ -1350,21 +1350,13 @@ spec:
...
-					_, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Update(context.Background(), vm, metav1.UpdateOptions{})
-					if err != nil {
-						return false
-					}
-					updatedVM, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Get(context.Background(), vmYaml.vmName, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					return updatedVM.Spec.Running == nil && updatedVM.Spec.RunStrategy != nil && *updatedVM.Spec.RunStrategy == runStrategyHalted
-				}, 30*time.Second, 3*time.Second).Should(BeTrue())
+				By("Changing run stategy to halted to be able to restore the vm")
+				patches, err := patch.New(patch.WithAdd("/spec/runStrategy", &runStrategyHalted)).GeneratePayload()
```
Comment (codingben) [inline tests/operator/operator.go:1354]: nit: Name it as `patchBytes`.

File: tests/operator/operator.go [hunk 2]
```diff
@@ -1350,21 +1350,13 @@ spec:
...
+				vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Patch(context.Background(), vmYaml.vmName, types.JSONPatchType, patches, metav1.PatchOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(vm.Spec.Running).ToNot(BeNil())
```
Comment (codingben) [inline tests/operator/operator.go:1357]: Why this assert is required, if we can check only `vm.Spec.RunStrategy`?

**Review by codingben — APPROVED** [review]
Thanks Ram! I think changes are okay, just please address my comments.

/lgtm

File: tests/operator/operator.go (hunk 2 above)
Comment (RamLavi) [reply tests/operator/operator.go:1357]: I think you're right. fixed.

File: tests/operator/operator.go (hunk 1 above)
Comment (RamLavi) [reply tests/operator/operator.go:1354]: DONE

**Comment by RamLavi** [comment]
[@change](https://github.com/kubevirt/kubevirt/compare/f07e30eb7901a735c7fdfc156222127cab6483b8..19c33b621638ef269c81475d7c69ad9a25c7c60a): Addressed @codingben review. PTAL

**Comment by RamLavi** [comment]
> [Change](https://github.com/kubevirt/kubevirt/compare/f07e30eb7901a735c7fdfc156222127cab6483b8..19c33b621638ef269c81475d7c69ad9a25c7c60a): Addressed @codingben review. PTAL

**Review by codingben — APPROVED** [review]
/lgtm

File: tests/operator/operator.go (hunk 2 above)
Comment (enp0s3) [reply tests/operator/operator.go:1357]: @RamLavi I think you didn't push the changes

**Review by enp0s3 — COMMENTED** [review]
@RamLavi Hi, looks great, please see a small comment below

**Comment by RamLavi** [comment]
> @RamLavi Hi, looks great, please see a small comment below

not sure what comment you refer to. Can you specify?

**Comment by RamLavi** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/19c33b621638ef269c81475d7c69ad9a25c7c60a..bbc6a9b84facb55fbead22ce9b580c0eb7f250ea): Rebase

**Comment by RamLavi** [comment]
/retest
Prow hiccup

**Comment by RamLavi** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/bbc6a9b84facb55fbead22ce9b580c0eb7f250ea..2fc1d1578e54dab7b74f6682ced95ebda1ade880): Rebase

**Comment by RamLavi** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/2fc1d1578e54dab7b74f6682ced95ebda1ade880..ecec3a35405bfc6a4ab6c281b56c88589bc5a558): Removed invalid exception.

**Review by codingben — APPROVED** [review]
/lgtm

**Comment by enp0s3** [comment]
/approve
@RamLavi Thank you!

**Pull request merged**